tools/screenshot-test/src/app/firebase.service.ts

```diff
@@ -0,0 +1,198 @@
+import {Injectable} from '@angular/core';
+import * as firebase from 'firebase';
+
+const request = require('request');
+const config = require('../config.json');
+
+
+export class ScreenshotResult {
+  /** PR information: the pull request number */
+  prNumber: string;
+  /** PR information: the sha of the pull request commit */
+  sha: string;
+  /** PR information: The travis job ID */
+  travis: string;
+
+  /** Test result, the test names */
+  testnames: string[];
+  /** Test result: passed or failed. The value can be true if test failed but PR approved by user */
+  result: boolean;
+  /** Test results: passed or failed for each test */
+  results: Map<string, boolean> = new Map<string, boolean>();
+  /**
+   * Test approved: whether the test images are copied to goldens.
+   * The value is the commit SHA and date/time of approval.
```

Comment:
Changed to `approved/${SHA}/${dateTime}` so we can approve multiple times. At first the value is a single boolean, and can be set only once for each PR which doesn't work when we have multiple commits and need to approve a commit more than once (for example when firebase functions failed).